src/som/vmobjects/SClass.java

```diff
@@ -63,14 +64,29 @@
   @CompilationFinal private ClassFactory instanceClassGroup; // the factory for this object
 
   protected final SObjectWithClass enclosingObject;
+  private MaterializedFrame context;
```

Comment:
Yes, it should be final. I've added a method to the `ClassInstantationNode` to enable this.